File: org/cooklang/Timer.java
package org.cooklang;

public class Timer implements DirectionItem {
    public String units;
    public String name;

    private Float quantityFloat;
    private String quantityString;

    public void setQuantityFloat(Float newQuantity) {
        this.quantityFloat = newQuantity;
    }

    public void setQuantityString(String newQuantity) {
        this.quantityString = newQuantity;
    }

    public void setUnits(String newUnits) {
        this.units = newUnits;
    }

    public void setName(String newName) {
        this.name = newName;
    }

    public String toString() {
        return String.format("%s %s %s %s", name, quantityFloat, quantityString, units);
    }
}

File: parser/src/main/java/org/cooklang/parser/Timer.kt
package org.cooklang.parser

import org.cooklang.parser.DirectionItem

data class Timer(
    var units: String? = null,
    var name: String? = null,
    var quantityFloat: Float? = null,
    var quantityString: String? = null
) : DirectionItem {
    override fun getDisplayString(): String {
        return "$name ${quantityFloat?: quantityString?: ""} ${units ?: ""}"
    }
}